### PR TITLE
Add case information to the form summary

### PR DESCRIPTION
**Pull request by proteusvacuum**
Gives you a combined view of your application in the form summary! 
Shows you which case properties are loaded into or saved from form questions. This is helpful for tracing dependencies within you app, and giving a good overview of how data flows through it.

For example, here you can see that all 3 of these questions load the `hiv_treatment_initiated` property from a `notification` case type and immediately save it out to both a `pregnancy_pp` and `client` case type. It would have been hard to see this type of data flow anywhere else in HQ. 

screenshot from 2019-02-06 22-32-52

Best to review :blowfish: / :blowfish: 

I'm also very open to suggestions on how best to present this data (currently I find the load and save icons a little subtle). 

# Update 14/2

We settled on this as the UI, pending user feedback:
screenshot from 2019-02-15 13-40-55

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/summary/form_summary.js [hunk 1]
```diff
@@ -11,10 +11,10 @@ hqDefine('app_manager/js/summary/form_summary',[
 ], function ($, _, ko, initialPageData, assertProperties, models, utils) {
     var formSummaryModel = function (options) {
         var self = models.contentModel(_.extend(options, {
-            query_label: gettext("Filter questions"),
+            query_label: gettext("Filter questions or cases"),
```
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/summary/form_summary.js]: Nitpick: "or case types"?

Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/summary/form_summary.js]: Nevermind, looks like you're searching both case type and case property.

File: corehq/apps/app_manager/static/app_manager/js/summary/form_summary.js [hunk 2]
```diff
@@ -25,6 +25,11 @@ hqDefine('app_manager/js/summary/form_summary',[
                             questionIsVisible = questionIsVisible || _.find(question.options, function (option) {
                                 return match(query, option.value + self.translateQuestion(option));
                             });
+                            var casePropsVisible = _.find(question.load_properties.concat(question.save_properties), function (prop) {
+                                return match(query, prop[0] + prop[1]);
```
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/summary/form_summary.js]: Nitpick: `match(query, prop[0]) || match(query, prop[1])` would be a little more semantic.

File: corehq/apps/reports/formdetails/readable.py [hunk 3]
```diff
@@ -170,6 +171,10 @@ class CaseTypeMeta(JsonObject):
     error = StringProperty()
     has_errors = BooleanProperty()
 
+    # store where this case type gets loaded so that we can look it up more easily later
+    load_properties = DictProperty()  # {"form_id:question_path" -> [CaseProperty, ...]}
```
Comment (orangejenny) [inline corehq/apps/reports/formdetails/readable.py]: I'm generally opposed to magically-formatted strings, even though this isn't used in many places. Thoughts on changing this to `{form_id -> {question_path -> [CaseProperty, ...]}}`?

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/reports/formdetails/readable.py (hunk 3 above)
Comment (proteusvacuum) [reply corehq/apps/reports/formdetails/readable.py]: Yeah, that is much better. I have no idea what I was thinking with that...

**Comment by orangejenny** [comment]
A couple of UI thoughts. Discard anything that isn't useful. 

1) It's hard to distinguish between loading and saving. Here's a form with one question that loads and one that saves:
screen shot 2019-02-12 at 4 52 13 pm

One thought I had is to use an arrow to show data "going into" or "coming out of" the case:
screen shot 2019-02-12 at 4 54 11 pm

Or like this:
screen shot 2019-02-12 at 4 55 19 pm

Both break the pattern of one icon followed by indeterminate text, but that might be be okay (or it might be terrible, I didn't test how it looks when there's a lot more going on).

If you want to keep one icon, maybe instead of upload/download icons, maybe "read"/"write" icons?

screen shot 2019-02-12 at 5 10 46 pm

or

screen shot 2019-02-12 at 5 11 02 pm

2) The grey labels are pretty visually heavy, even on the small app in the screenshot, and also repetitive. I imagine that if you display case properties at the same time as calculations, etc., the case properties are going to be very dominant.

What if the case type were visually lighter than the surrounding text, instead of heavier? Here's a form:
screen shot 2019-02-12 at 5 20 10 pm

And here's that form where case types are `.text-muted` instead of `.label.label-default`:
screen shot 2019-02-12 at 5 20 58 pm

I don't like that only the color differentiates them, but I do like that case type fades more into the background. The case type could also be monospace font, or some other treatment so it'd be more distinct from the property name but wouldn't dominate the property name.

On a different but related subject, what would you think of marking the *form* with the case type, and then for questions just displaying the property? Forms that deal with multiple case types would still need the per-question labels, and I'm not sure how complicated the average GS app is, so I don't think this would be worth doing if it'd only affect a small fraction of heavy app summary users, just something to think about.

**Comment by orangejenny** [comment]
One last thought - on the images above with the arrows & briefcase icons, replacing the briefcase with a form (or a question mark) might make more sense, so the icon represents the question itself and then the arrow shows it going into, or coming out of, the case property. The briefcase + arrow + property seems confusing now that I look at it.

**Review by orangejenny — APPROVED** [review]

**Comment by kaapstorm** [comment]
**This is awesome!!!**

Some opinions on UI: 
* The Save and Load icons are OK. But I agree with @orangejenny that the Case + Arrow icons are clearer. (Maybe Case + Right Arrow for "load", and Case + Left Arrow for "save"?) I find the "eye" icon a little jarring, I don't know why. I personally prefer the book icon, but would lean towards an [open book][book], which is so much more readable than a closed one. :grin:  
* I do like the idea of `.text-muted`. To make it clearer on crappy monitors, or for users who have difficulty seeing the difference in tone, we could also try adding a colon after the case type.

  [book]: https://fontawesome.com/icons/book-open?style=solid

**Review by kaapstorm — APPROVED** [review]

**Comment by proteusvacuum** [comment]
We settled on this: 
screenshot from 2019-02-14 14-24-52
Updating PR description with that screenshot

**Review by orangejenny — APPROVED** [review]

**Pull request merged**